src/Uno.UI/Extensions/TimeSpanExtensions.cs

```diff
@@ -75,5 +75,28 @@ internal static TimeSpan RoundToMinuteInterval(this TimeSpan time, int interval)
 
 			return time;
 		}
+
+		/// <summary>
+		/// Normalize timespan in the specified range
+		/// </summary>
+		/// <param name="minValue">minimum value of timespan</param>
+		/// <param name="maxValue">maximum value of timespan</param>
+		/// <returns>Normalized timespan</returns>
+		internal static TimeSpan Normalize(this TimeSpan value, TimeSpan minValue, TimeSpan maxValue)
```

Comment:
As discussed, seems better to remove the min/max and just limit to the "day" use case